src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java

```diff
@@ -0,0 +1,193 @@
+                                  rptId,
+                                  p.paymentToken()))
+                  : nodoOperations
+                      .activatePaymentRequest(partialPaymentRequestInfo, newTransactionRequestDto)
+                      .doOnSuccess(
+                          p ->
+                              log.info(
+                                  "Nodo activation for {} with paymentToken {}",
+                                  rptId,
+                                  p.paymentToken()));
+            })
+        .doOnNext(
+            paymentRequestInfo -> {
+              log.info(
+                  "Cache Nodo activation info for {} with paymentToken {}",
+                  rptId,
+                  paymentRequestInfo.paymentToken());
+              paymentRequestsInfoRepository.save(paymentRequestInfo);
+            })
+        .flatMap(
+            paymentRequestInfo -> {
+              final String transactionId = UUID.randomUUID().toString();
+              final SessionRequestDto sessionRequest =
+                  new SessionRequestDto()
+                      .email(newTransactionRequestDto.getEmail())
+                      .rptId(paymentRequestInfo.id().value())
+                      .transactionId(transactionId);
+
+              return ecommerceSessionsClient
+                  .createSessionToken(sessionRequest)
+                  .map(sessionData -> Tuples.of(sessionData, paymentRequestInfo));
+            })
+        .flatMap(
+            args -> {
+              final SessionDataDto sessionDataDto = args.getT1();
+              final PaymentRequestInfo paymentRequestInfo = args.getT2();
+
+              return Mono.just(paymentRequestInfo.paymentToken())
+                  .flatMap(
+                      paymentToken ->
+                          isValidPaymentToken(paymentToken)
+                              ? Mono.just(Tuples.of(
+                                  newTransactionActivatedEvent(
+                                      paymentRequestInfo.amount(),
+                                      paymentRequestInfo.description(),
+                                      sessionDataDto.getEmail(),
+                                      sessionDataDto.getTransactionId(),
+                                      sessionDataDto.getRptId(),
+                                      paymentToken),
+                                  null))
+                              : Mono.just(Tuples.of(
+                                  null,
+                                  newTransactionActivationRequestedEvent(
+                                      paymentRequestInfo.amount(),
+                                      paymentRequestInfo.description(),
+                                      sessionDataDto.getEmail(),
+                                      sessionDataDto.getTransactionId(),
+                                      sessionDataDto.getRptId()))));
```

Comment:
It could be worth to replace in a future PR `Mono<TransactionActivatedEvent, TransactionActivationRequestedEvent>` with an `Mono<Either<TransactionActivatedEvent, TransactionActivationRequestedEvent>>` (e.g. from vavr.io)